cmd/kubeadm/app/preflight/checks.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preflight
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+
+	"k8s.io/kubernetes/pkg/util/initsystem"
+)
+
+type PreFlightError struct {
+	Msg   string
+	Count int
+}
+
+func (e *PreFlightError) Error() string {
+	return fmt.Sprintf("preflight check error\n  count: %d \n  msg: %s", e.Count, e.Msg)
+}
+
+// PreFlightCheck validates the state of the system to ensure kubeadm will be
+// successful as often as possilble.
+type PreFlightCheck interface {
+	Check() (warnings, errors []error)
+}
+
+// ServiceCheck verifies that the given service is enabled and active. If we do not
+// detect a supported init system however, all checks are skipped and a warning is
+// returned.
+type ServiceCheck struct {
+	service string
+}
+
+func (sc ServiceCheck) Check() (warnings, errors []error) {
+
+	initSystem := initsystem.GetInitSystem()
+	if initSystem == nil {
+		return []error{fmt.Errorf("no supported init system detected, skipping service checks for %s", sc.service)}, nil
+	}
+
+	warnings = []error{}
+
+	if !initSystem.ServiceExists(sc.service) {
+		warnings = append(warnings, fmt.Errorf("%s service does not exist", sc.service))
+		return warnings, nil
+	}
+
+	if !initSystem.ServiceIsEnabled(sc.service) {
+		warnings = append(warnings,
+			fmt.Errorf("%s service is not enabled, please run 'systemctl enable %s.service'",
+				sc.service, sc.service))
+	}
+
+	if !initSystem.ServiceIsActive(sc.service) {
+		errors = append(errors,
+			fmt.Errorf("%s service is not active, please run 'systemctl start %s.service'",
+				sc.service, sc.service))
+	}
+
+	return warnings, nil
```

Comment:
I knew I was seeing something weird here, I failed to return the errors list we build up.